packages/next/next-server/server/render.tsx

```diff
@@ -379,6 +379,33 @@ export async function renderToHTML(
       const data = await unstable_getStaticProps!({
         params: isDynamicRoute(pathname) ? query : undefined,
       })
+
+      const invalidKeys = Object.keys(data).filter(
+        key => key !== 'revalidate' && key !== 'props'
+      )
+
+      if (invalidKeys.length) {
+        throw new Error(
+          `Additional keys were returned from \`getStaticProps\`. Properties intended for your component must be nested under the \`props\` key, e.g.:\n\n\treturn { props: { title: 'My Title', content: '...' }\n\nKeys that need moved: ${invalidKeys.join(
+            ', '
+          )}.
+        `
+        )
+      }
+
+      if (typeof data.revalidate === 'number') {
+        if (data.revalidate < 0) {
+          throw new Error(
+            `A page's revalidate can not be less than 0. If you want to always revalidate your page use getInitialProps instead of getStaticProps.`
+          )
+        } else if (data.revalidate > 31536000) {
+          // if it's greater than a year for some reason error
+          throw new Error(
+            `A page's revalidate can not be set to more than a year. If you want to prevent revalidating you can use \`false\` instead.`
```

Comment:
I think this should be a warning. Also, let's adjust the text:
```suggestion
            `Warning: A page's revalidate option was set to more than a year. This may have been done in error.\nTo only run getStaticProps at build-time and not revalidate at runtime, you can set \`revalidate\` to \`false\`!`
```